code/lib/preview-api/src/modules/store/csf/prepareStory.test.ts

```diff
@@ -704,3 +704,44 @@ describe('moduleExport', () => {
     expect(story.moduleExport).toBe(storyObj);
   });
 });
+
+describe('prepareMeta', () => {
+  it('returns the same as prepareStory', () => {
+    const meta = {
+      id,
+      title,
+      moduleExport,
+      tags: ['some-tag'],
+      parameters: {
+        a: { name: 'component' },
+        b: { name: 'component' },
+        nested: { z: { name: 'component' }, y: { name: 'component' } },
+      },
+      args: {
+        a: 'component',
+        b: 'component',
+        nested: { z: 'component', y: 'component' },
+      },
+      argTypes: {
+        a: { name: 'a-story', type: booleanType },
+        nested: { name: 'nested', type: booleanType, a: 'story' },
+      },
+    };
+    const preparedStory = prepareStory({ id, name, moduleExport }, meta, { render });
+    const preparedMeta = prepareMeta(meta, { render });
+
+    expect(preparedMeta).toMatchObject({
+      ...preparedStory,
+      // properties that are actually different between prepareMeta and prepareStory
+      moduleExport: undefined,
+      id: 'id--__meta',
```

Comment:
We shouldn't set an `id`

Reply:
I'm not sure about that. `componentAnnotations` have `id`s, and all of the tests for `prepareStory` pass in an id as the `componentAnnotations` id.

I agree that we shouldn't set a fake one, we should just use the one from `componentAnnotations`, I've changed that.